be/src/exprs/cast_expr_struct.cpp

```diff
@@ -0,0 +1,131 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "column/array_column.h"
+#include "column/column_builder.h"
+#include "column/column_helper.h"
+#include "column/column_viewer.h"
+#include "column/json_column.h"
+#include "exprs/cast_expr.h"
+#include "exprs/expr_context.h"
+#include "gutil/casts.h"
+#include "gutil/strings/split.h"
+#include "gutil/strings/strip.h"
+#include "gutil/strings/substitute.h"
+#include "jsonpath.h"
+#include "runtime/memory/memory_resource.h"
+#include "types/logical_type.h"
+#include "util/slice.h"
+#include "velocypack/Iterator.h"
+
+namespace starrocks {
+
+#define APPEND_NULL(json_columns, null_column) \
+    for (auto& json_column : json_columns) {   \
+        json_column.append_null();             \
+    }                                          \
+    null_column->append(1);
+
+StatusOr<ColumnPtr> CastJsonToStruct::evaluate_checked(ExprContext* context, Chunk* input_chunk) {
+    ASSIGN_OR_RETURN(ColumnPtr column, _children[0]->evaluate_checked(context, input_chunk));
+    if (column->only_null()) {
+        return ColumnHelper::create_const_null_column(column->size());
+    }
+
+    ColumnViewer<TYPE_JSON> src(column);
+    NullColumn::Ptr null_column = NullColumn::create();
+
+    // 1. Cast Json to json columns.
+    size_t field_size = _type.children.size();
+    DCHECK_EQ(field_size, _type.field_names.size());
+    Columns columns;
+    vector<ColumnBuilder<TYPE_JSON>> json_columns;
+    for (size_t i = 0; i < field_size; i++) {
+        ColumnPtr field_column = ColumnHelper::create_column(
+                _type.children[i], true, false, src.size());
+        columns.emplace_back(field_column);
+        ColumnBuilder<TYPE_JSON> json_column_builder(src.size());
+        json_columns.emplace_back(json_column_builder);
+    }
+    for (size_t i = 0; i < src.size(); i++) {
+        if (src.is_null(i)) {
+            APPEND_NULL(json_columns, null_column);
+            continue;
+        }
+        const JsonValue* json_value = src.value(i);
+        if (json_value && json_value->get_type() == JsonType::JSON_ARRAY) {
+            vpack::Slice json_slice = json_value->to_vslice();
+            DCHECK(json_slice.isArray());
+            size_t index = 0;
+            for (const auto& element : vpack::ArrayIterator(json_slice)) {
+                JsonValue element_value(element);
+                json_columns[index].append(std::move(element_value));
+                index++;
+            }
+            // The element number of the json array should exactly match the field number of the struct.
+            if (index != columns.size()) {
+                APPEND_NULL(json_columns, null_column);
+                continue ;
+            }
+            null_column->append(0);
```

Comment:
the elements of array exactlty mapping to struct field ? I think it's difficult to use....

Reply:
and the index handle is error? when index > subfiled size? and when index < subfield, all json_columns need null?

Reply:
It could return result even if the sizes do not match, but the result might be confused.
e.g. [1, 2, 3] -> struct <int,int,int,int,int>
The result will be 1, 2, 3, null, null. The last two nulls are confused.